### PR TITLE
SpinButton: Events not firing #3282 (#4661)

### PR DESCRIPTION
**Pull request by lijunle**
* Fixing: Events not firing #3282
* Added tests for onIncrement, onDecrement and onValidate
* Adding change file.

This fix is ported from fabric@6 to fabric@5.

Related: #4661

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.test.tsx [hunk 1]
```diff
@@ -643,4 +641,46 @@ describe('SpinButton', () => {
...
+
+    ReactTestUtils.Simulate.mouseDown(buttonDOM,
+      {
+        type: 'mousedown',
+        clientX: 0,
+        clientY: 0
+      }
+    );
+
+    expect(onDecrement).toBeCalled();
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.test.tsx]: Consider testing that the callback is only invoked _once_ per action using [toHaveBeenCalledTimes](https://jest-bot.github.io/jest/docs/expect.html#tohavebeencalledtimesnumber). Can also be applied to the `onValidate` test.

Regardless of if you choose to swap assertions, thank you for adding test coverage.

File: common/changes/office-ui-fabric-react/master_2018-05-01-00-32.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "SpinButton: addressing onValidate, onIncrement, and onDecrement callbacks to be called correctly.",
+      "type": "patch"
+    }
+  ],
+  "packageName": "office-ui-fabric-react",
+  "email": "dzearing@microsoft.com"
```
Comment (KevinTCoughlin) [inline common/changes/office-ui-fabric-react/master_2018-05-01-00-32.json]: Is this the correct email?

**Comment by lijunle** [comment]
@KevinTCoughlin This is [a port](https://github.com/OfficeDev/office-ui-fabric-react/pull/4661) from fabric@6 to fabric@5. Do we fix the code separately for v5 branch?

**Comment by lijunle** [comment]
@dzearing @micahgodbolt @KevinTCoughlin Could you please help to merge this PR into v5 branch? Thanks!

**Pull request merged**